### DISCUSSION TITLE
trying to filter out specific things and setting up preferences

### DISCUSSION THREAD

**Discussion by epross672** [Flexget/Flexget#4050] [General]
I am trying to set something up where i look for specific things and then grab the 2nd preference, etc.

however

`tv4:
    retry_failed:
      retry_time: 20 minutes
      retry_time_multiplier: 2
      max_retries: 100

    verify_ssl_certificates: no
    thetvdb_lookup: yes

    if:
      # First preference: 1080p AMZN with [TGx]
      - "'1080P' in title and 'AMZN' in title and '[TGx]' in title":
          accept: yes  # Always download if found

      # Second preference: 1080p with [TGx]
      - "'1080P' in title and '[TGx]' in title":
          delay: 12 hours  # Allow download only after delay if the first isn't found

      # Third preference: 1080p anything
      - "'1080P' in title":
          delay: 24 hours  # Delay to give time for better options to show up

      # Rescan older options for the best available
      - "'1080P' in title and 'AMZN' in title and '[TGx]' in title and not seen_info_hash":
          accept: yes  # Accept best option if [TGx] and AMZN appear with 1080P and not previously seen
      - "'1080P' in title and 'AMZN' in title and not seen_info_hash":
          accept: yes  # Accept if only the AMZN option is available and not previously seen
      - "'1080P' in title and '[TGx]' in title and not seen_info_hash":
          accept: yes  # Accept if only the TGx option is available and not previously seen`

but i am getting:

` [/templates/tv2/if/3/'1080P' in title and 'AMZN' in title and '[TGx]' in title and not seen_info_hash] {'accept': True} is not valid under any of the given schemas
2024-09-28 10:50:43 CRITICAL manager                       [/templates/tv2/if/4/'1080P' in title and 'AMZN' in title and not seen_info_hash] {'accept': True} is not valid under any of the given schemas
`

wondering if i am handling this wrong...

**Comment by BrutuZ** [comment]
- Use `regexp` for the first one instead of `if`
- Add mutual exclusions to the second and third (resolution in title source not in title...)
- `seen_info_hash` is its own filter that rejects entries, you can't do that last block

**Comment by epross672** [comment]
What I am trying to do is always get the 

 "'1080P' in title and 'AMZN' in title and '[TGx]' in title":

Even if there's a 1080p AMZN first.... 

So if a normal 1080p one comes out first, and then 2 days later a 1080 amzn comes out, i want that.... but if 2 days after that, 1080 amzn tgx comes out, i want that.

I wonder if the delay will not work the way i want becuse it wont rescan will it?

**Comment by BrutuZ** [comment]
Maybe try something like this?
```yaml
accept_all: yes
quality: 1080p  # Also handles variants like FullHD
delay: 1 day

if:
  # First preference: 1080p AMZN with [TGx]
  - "'AMZN' in title and '[TGx]' in title": delay: 5 seconds  # Virtually now

  # Second preference: 1080p with [TGx]
  - "'[TGx]' in title and 'AMZN' not in title": delay: 12 hours
```

It will probably download duplicates though. Rescan as you expect is not a thing on its own, entries are either ignored when seen again or treated as if it's always the first time (by disabling the `seen`-related plugins for the task)
